### PR TITLE
Fixes #18303: Calling the relay-api to trigger a remote run on all nodes do an infinite loop on root server, killing it

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/18303

### CODE REVIEW

File: relay/sources/relayd/src/data/node.rs [hunk 1]
```diff
@@ -445,13 +457,24 @@ mod tests {
...
+            Ok(None)
+        );
+    }
+
     #[test]
     fn it_gets_sub_relays() {
-        let mut reference = vec![
-            "node1.rudder.local",
-            "node2.rudder.local",
-            "server.rudder.local",
```
Comment (VinceMacBuche) [inline relay/sources/relayd/src/data/node.rs:453]: Haha

File: relay/sources/relayd/src/data/node.rs [hunk 2]
```diff
@@ -237,15 +237,27 @@ impl NodesList {
...
 
+    /// Get all directly connected downstream policy servers
     pub fn my_sub_relays(&self) -> Vec<Host> {
         let mut relays = HashSet::new();
         for policy_server in self
             .list
             .data
             .values()
-            .filter_map(|v| self.list.data.get(&v.policy_server))
-            .filter(|v| v.policy_server == self.my_id)
```
Comment (ncharles) [inline relay/sources/relayd/src/data/node.rs:247]: we had the same code in the scala part in the past, with same consequences :)

**Review by ncharles — APPROVED** [review]

File: relay/sources/relayd/src/data/node.rs (hunk 1 above)
Comment (amousset) [reply relay/sources/relayd/src/data/node.rs:453]: funny, isn't it?

Comment (amousset) [reply relay/sources/relayd/src/data/node.rs:453]: unit tests at their best

**Pull request merged**